application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java

```diff
@@ -0,0 +1,335 @@
+     * <p>
+     * The checks include:
+     * <ul>
+     * <li>the role does not exist on the guild</li>
+     * <li>the target is not member of the guild</li>
+     * <li>the bot or author do not have enough permissions to interact with the target</li>
+     * <li>the bot or author do not have enough permissions to interact with the role</li>
+     * <li>the author does not have the required role for this interaction</li>
+     * <li>the bot does not have the MANAGE_ROLES permission</li>
+     * <li>the given reason is too long</li>
+     * </ul>
+     *
+     * @param role the role to change, or {@code null} if it does not exist on the guild
+     * @param actionVerb the interaction as verb, for example {@code "mute"} or {@code "unmute"}
+     * @param target the target user to change roles from, or {@code null} if the user is not member
+     *        of the guild
+     * @param bot the bot executing this interaction
+     * @param author the author attempting to interact with the target
+     * @param guild the guild this interaction is executed on
+     * @param hasRequiredRole a predicate used to identify required roles by their name
+     * @param reason the reason for this interaction
+     * @param event the event used to respond to the user
+     * @return Whether the bot and the author have enough permission
+     */
+    @SuppressWarnings({"MethodWithTooManyParameters", "BooleanMethodNameMustStartWithQuestion",
+            "squid:S107"})
+    static boolean handleRoleChangeChecks(@Nullable Role role, @NotNull String actionVerb,
+            @Nullable Member target, @NotNull Member bot, @NotNull Member author,
+            @NotNull Guild guild, @NotNull Predicate<? super String> hasRequiredRole,
+            @NotNull CharSequence reason, @NotNull Interaction event) {
+        if (role == null) {
+            event
+                .reply("Can not %s the user, unable to find the corresponding role on this server"
+                    .formatted(actionVerb))
+                .setEphemeral(true)
+                .queue();
+            logger.warn("The guild '{}' does not have a role to {} users.", guild.getName(),
+                    actionVerb);
+            return false;
+        }
+
+        // Member doesn't exist if attempting to change roles of a user who is not part of the guild
+        // anymore.
+        if (target == null) {
+            handleAbsentTarget(actionVerb, event);
+            return false;
+        }
+        if (!handleCanInteractWithTarget(actionVerb, bot, author, target, event)) {
+            return false;
+        }
+        if (!handleCanInteractWithRole(bot, author, role, event)) {
+            return false;
+        }
+        if (!handleHasAuthorRole(actionVerb, hasRequiredRole, author, event)) {
+            return false;
+        }
+        if (!handleHasBotPermissions(actionVerb, Permission.MANAGE_ROLES, bot, guild, event)) {
+            return false;
+        }
```

Comment:
This can be simplified to 
```suggestion
        if (!handleCanInteractWithTarget(actionVerb, bot, author, target, event) || !handleCanInteractWithRole(bot, author, role, event) || !handleHasAuthorRole(actionVerb, hasRequiredRole, author, event) || (!handleHasBotPermissions(actionVerb, Permission.MANAGE_ROLES, bot, guild, event)) {
            return false;
        }
      
```

Reply:
Issue is no-one can read that, I'd say keep it separate.

Reply:
I kept it separate for readability, yes. The combined statement could be made readable if spotless would format it differently (one condition per line) but since it doesnt do that, I would rather have it separate. (Also keeps the style from all the other commands that did it the same way).